lms/djangoapps/certificates/api.py

```diff
@@ -26,7 +26,8 @@
 log = logging.getLogger("edx.certificate")
 
 
-def generate_user_certificates(student, course_key, course=None, insecure=False, generation_mode='batch'):
+def generate_user_certificates(student, course_key, course=None, insecure=False, generation_mode='batch',
```

Comment:
course argument is allowed to avoid db hit. If context already has course available it can be passed to this course to avoid fetching it. If course is not present course_key would be used to fetch course.